bundles/core/src/main/java/com/adobe/cq/forms/core/components/internal/services/formsportal/OpenDraftOperation.java

```diff
@@ -89,12 +90,17 @@ public OperationResult execute(Map<String, Object> parameterMap) {
             .collect(Collectors.toMap(Map.Entry::getKey, v -> (String[]) v.getValue()));
         String modelID = Arrays.stream(Objects.requireNonNull(map.get(OPERATION_MODEL_ID))).findFirst().orElse(null);
         try {
-            DraftModel dm = draftService.getDraft(modelID);
-            String formPath = GuideUtils.convertFMAssetPathToFormPagePath(GuideUtils.convertGuideContainerPathToFMAssetPath(dm
-                .getFormPath()));
-            String formLink = String.format(DRAFT_LINK, formPath, modelID, suffix);
-            result.put("formLink", formLink);
-            result.put("status", "success");
+            if (draftService != null) {
+                DraftModel dm = draftService.getDraft(modelID);
+                String formPath = GuideUtils.convertFMAssetPathToFormPagePath(GuideUtils.convertGuideContainerPathToFMAssetPath(dm
+                    .getFormPath()));
+                String formLink = String.format(DRAFT_LINK, formPath, modelID, suffix);
+                result.put("formLink", formLink);
+                result.put("status", "success");
+            } else {
+                LOGGER.error("DraftService is not available to fetch draft with id " + modelID);
```

Comment:
Can we explicitly log that why is the service not available (ie) upgrade to SP24 ?